presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -213,6 +235,18 @@ private SqlQueryScheduler(
 
         OutputBufferId rootBufferId = getOnlyElement(rootOutputBuffers.getBuffers().keySet());
         sectionedPlan = extractStreamingSections(plan);
+
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.nodeScheduler = requireNonNull(nodeScheduler, "nodeScheduler is null");
+        this.remoteTaskFactory = requireNonNull(remoteTaskFactory, "remoteTaskFactory is null");
+        this.splitSourceFactory = requireNonNull(splitSourceFactory, "splitSourceFactory is null");
+        this.session = requireNonNull(session, "session is null");
+        this.splitBatchSize = requireNonNull(splitBatchSize, "splitBatchSize is null");
+        this.nodePartitioningManager = requireNonNull(nodePartitioningManager, "nodePartitioningManager is null");
+        this.queryExecutor = requireNonNull(queryExecutor, "queryExecutor is null");
+        this.schedulerExecutor = requireNonNull(schedulerExecutor, "schedulerExecutor is null");
+        this.failureDetector = requireNonNull(failureDetector, "failureDetector is null");
+        this.nodeTaskMap = requireNonNull(nodeTaskMap, "nodeTaskMap is null");
         List<StageExecutionAndScheduler> stageExecutions = createStageExecutions(
```

Comment:
With a lot of fields added into `SqlQueryScheduler`,  I think the signature of this method can be simplified since many of the parameters are now fields now so don't need to be passed in?

Reply:
yes, working on that now :).